src/D2L.Bmx/Authenticate.cs

```diff
@@ -55,18 +60,61 @@ internal class Authenticator {
 				PromptMfaInput( selectedMfa.Name );
 				throw new NotImplementedException();
 			}
-			authenticatedState = await oktaApi.AuthenticateChallengeMfaOktaAsync( authState, selectedMfaIndex - 1, mfaInput );
+			sessionToken = await oktaApi.AuthenticateChallengeMfaOktaAsync( authState, selectedMfaIndex - 1, mfaInput );
 		} else if( authState.OktaSessionToken is not null ) {
-			authenticatedState = new OktaAuthenticatedState( true, authState.OktaSessionToken );
+			sessionToken = authState.OktaSessionToken;
 		}
 
-		if( authenticatedState is not null ) {
-			return authenticatedState;
+		if( !sessionToken.Equals( "" ) ) {
```

Comment:
nit: prefer ``string.IsNullOrEmpty`` for these kind of checks to see if a string is set

Reply:
I don't think we'll have a scenario where it's null but I've swapped to that and for the other times it's used

Reply:
Yep, its mostly for consistency with how we write empty string checks with the other projects in the team